draft-gpew-priv-ppm.md

```diff
@@ -629,268 +653,638 @@ encoded value of the following form:
+~~~
+{: #aggregation-flow-helper-state title="Helper's state machine while preparing
+a report share for aggregation."}
+
+Upon receiving a ReportShare from the leader, the helper initializes its
+preparation state as described in {{prep-start}} (denoted by `prep_start` in
+{{aggregation-flow-helper-state}}). It then executes the following procedure
+until it moves to a terminal state:
+
+* If failed, then send the leader the Transition message and move to FAILED.
+* If finished, then send the leader the Transition message, move to FINISHED,
+  and proceed as described in {{out-to-agg-share}}.
+* If continued, then send the leader the Transition message and move to WAITING.
+
+Upon receiving a Transition message from the leader, proceed as follows. If the
+leader continued, then update the prepare state and compute the next Transition
+message as described below. Otherwise, if the leader finished or failed, then
+move to INVALID and proceed as described in {{aggregate-request}}.
+
+The helper computes its next state transition (denoted `prep_next`) as
+
+~~~
+out = prep_state.next(payload)
+~~~
+
+where `payload` is the previous VDAF message sent by the leader. If this
+operation fails, then the helper fails with error `vdaf-prep-error`. Otherwise,
+it interprets `out` as follows. If this is the last round of VDAF preparation
+phase, then `out` is the helper's output share, in which case the helper
+finishes. Otherwise, the helper continues with `out` as its next VDAF message.
+
+### Aggregating Output Shares {#out-to-agg-share}
+
+Once processing of a report share reaches the FINISHED state, the aggregator
+stores the the recovered output share until the batch to which it pertains is
+collected. To aggregate the output shares, the aggregator runs the aggregation
+algorithm specified by the VDAF:
+
+~~~
+agg_share = output_to_aggregate_shares(agg_param,
+                                       output_shares)
+~~~
+
+Note that for most VDAFs, it is possible to aggregate output shares as they
+arrive rather than wait until the batch is collected. To do so however, it is
+necessary to enforce the batch parameters as described in
+{{batch-parameter-validation}} so that the aggregator knows which aggregate
+share to update.
+
+The leader needs to store a report for as long as it can be collected. Once a
+report has been collected `max_batch_lifetime` times, the leader SHOULD remove
+the report from storage.
+
+### Aggregate Messages {#aggregate-message-auth}
+
+The following messages are defined in this section and are used to aggregate a
+number of reports simultaneously:
+
+~~~
+opaque AggregationJobID[32];
+
+enum {
+  agg_init_req(0),
+  agg_req(1),
+} AggregateReqType;
 
 struct {
-  Nonce nonce;
-  opaque message<0..2^16-1>; // VDAF message
-} AggregateSubResp;
+    AggregateReqType msg_type;
+    select (msg_type) {
+        agg_init_req: AggregateInitReq;
+        agg_req: AggregateReq;
+    }
+    opaque tag[32];
+} AggregateReq;
```

Comment:
There are currently two structures called `AggregateReq`: this one, and another defined on L1176 (which is embedded in this message if `msg_type == aqq_req`).

My strawman suggestion would be to keep this one named `AggregateReq`, and name the sub-message `AggregateContinueReq`. TBH I'm not super-happy with that name, maybe something better is apparent -- I don't care too much as long as they're named separately.